src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,65 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
 public class WorkWithFile {
+    private static final String OPERATION_TYPE_BUY = "buy";
+    private static final String OPERATION_TYPE_SUPPLY = "supply";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        List<String> records = new ArrayList<>();
+        StringBuilder report = new StringBuilder();
+        readFromFile(fromFileName, records, report);
+        writeToFile(toFileName, report);
```

Comment:
This method is doing too much: it's reading from a file, processing the data, and writing to another file. Consider splitting it into separate methods to adhere to the principle that each method should be responsible for one task.